deeprank/features/Edesolv.py

```diff
@@ -0,0 +1,411 @@
+                #Calculate atom complex edesolv, free edesolv and final edesolv (cpx - free)
+                atom.esolcpx = atom.cpx_sasa * atom.solv
+                atom.esolfree = atom.free_sasa * atom.solv
+                atom.edesolv = atom.esolcpx - atom.esolfree
+                esolcpx += atom.esolcpx
+                esolfree += atom.esolfree
+                
+                # xyz-val
+                # { (0|1,x,y,z) : [val] }
+                chain = [{chainA:0,chainB:1}[key[1]]]
+                k = tuple(chain + coords)
+
+                #Human readable
+                self.edesolv_data[(key[1], key[3], key[2], key[4])] = [atom.edesolv]
+                #XYZ data
+                self.edesolv_data_xyz[k] = [atom.edesolv]
+
+        self.feature_data['Edesolv'] = self.edesolv_data
+        self.feature_data_xyz['Edesolv'] = self.edesolv_data_xyz
+
+##################################################################
+
+##################################################################
+
+def __compute_feature__(pdb_data, featgrp, featgrp_raw):
+    """Main function called in deeprank for the feature calculations.
+
+    Args:
+        pdb_data (list(bytes)): pdb information
+        featgrp (str): name of the group where to save xyz-val data
+        featgrp_raw (str): name of the group where to save human readable data
+    """
+
+    edesolv_feat = Edesolv(pdb_data)
+    edesolv_feat.get_feature()
+
+    # export in the hdf5 file
+    edesolv_feat.export_dataxyz_hdf5(featgrp)
+    edesolv_feat.export_data_hdf5(featgrp_raw)
+
+    # close
+    edesolv_feat.db._close()        
+        
+class Atom():
+
+    def __init__(self,atom_specs):
+        """Atom class necessary to calculate atomic-level surface area,
+        solvation parameters and desolvation energy.
+        
+        Args:
+            atom_specs("""
+
+        serial = atom_specs[0]
+        chainID = atom_specs[1]
+        resn = atom_specs[2]
+        resid = atom_specs[3]
+        atom = atom_specs[4]
+
+        self.serial = serial
+        self.chainID = chainID
+        self.resn = resn
+        self.resid = resid
+        self.name = atom
+        if atom in ['C', 'O', 'N', 'CA']:
+            self.position = 'BB'
+        else:
+            self.position = 'SC'
+
+
+def Assign_solv_param(atom, disulf_bond=False):
+    """Assigns solvation parameter to an Atom object
+
+    Args:
+        atom (Atom): Edesolv.Atom object.
+        disulf_bond (bool, optional): Disulfide bond flag. If true, the atom is a sulfide involved in a disulfie bond. Defaults to False.
+
+    Returns:
+        None.
+
+    """
+    
+    arofac = 6.26
+    alifac = 1.27
+    polfac = 2.30
+
+
+    if atom.position == "BB" or atom.position=="SC": #Check meaning and position of this line
+        atom.solv = 0.0000
+
+    if (((atom.name.startswith('CG')) or (atom.name.startswith('CD')) or (atom.name.startswith('CE')) or (atom.name.startswith('CH')) or (atom.name.startswith('CZ')))
```

Comment:
We are using a line length of 80 characters, please follow this rule and split a long line to short ones.